File: 3609.minimum-moves-to-reach-target-in-grid.py
#
# @lc app=leetcode id=3609 lang=python3
#
# [3609] Minimum Moves to Reach Target in Grid
#

# @lc code=start
class Solution:
    def minMoves(self, sx: int, sy: int, tx: int, ty: int) -> int:
        moves = 0
        while tx > sx and ty > sy:
            if tx > ty:
                tx %= ty
            else:
                ty %= tx
            moves += 1
        if (tx == sx and (ty - sy) % tx == 0) or (ty == sy and (tx - sx) % ty == 0):
            return moves + max((ty - sy) // tx, (tx - sx) // ty)
        return -1
# @lc code=end

File: 3609.minimum-moves-to-reach-target-in-grid.cpp
#
# @lc app=leetcode id=3609 lang=cpp
#
# [3609] Minimum Moves to Reach Target in Grid
#
# @lc code=start
class Solution {
public:
    int minMoves(int sx, int sy, int tx, int ty) {
        int moves = 0;
        while (tx > sx && ty > sy) {
            if (tx > ty) {
                tx -= max(1, tx - sx);
            } else {
                ty -= max(1, ty - sy);
            }
            moves++;
        }
        return (tx == sx && ty == sy) ? moves : -1;
    }
};
# @lc code=end